File: prime.js
#!/usr/bin/env node

moment = require('moment');

function primes(n){
    // sieve of Eratosthenes
    var f = new Array();
    for (var i = 0; i <= n; i++) {
        f[i] = i % 2;
    }
    nn = Math.sqrt(n);
    for (var i = 3; i <= nn; i=i+2) {
        if (f[i] == 1) {
            for (var j=i*i; j<=n; j=j+2*i) {
                f[j]=0;
            }
        }
    }
    var p = [2];
    for (var i = 3; i <= n; i=i+2) {
        if (f[i]==1) {
            p.push(i);
        }
    }
    return p
}

t1 = moment().seconds() + moment().milliseconds()/1000;
n = 10000000;   //100以内的素数
P = primes(n);
t2 = moment().seconds() + moment().milliseconds()/1000;

console.log(t2 - t1)

File: prime.nim
#!/usr/bin/env nim
# nim r Programming/Others/Benchmarks/prime.nim

import std/math
import timeit

proc primes(): seq[int] =
    # sieve of Eratosthenes
    const n = 1000000
    var f: array[1..n, bool]
    for i in 1..n:
        f[i] = (i mod 2) == 1
    var i: int = 3
    var nn : int = int(floor(sqrt(float(n))))
    while i <= nn:
        if f[i]:
            var j :int = i^2
            while j<=n:
                f[j] = false
                j = j+ 2*i
        i = i + 2
 
    var P: seq[int]
    P = @[2]
    for x in countup(3, n, 2):
        if f[x]:
            P.add x
    return P
 

timeOnce("timeit"):
    discard primes()
